tests/e2e/project.spec.ts

```diff
@@ -0,0 +1,337 @@
+  ).toContainText(`${params.contributors} contributor(s)`);
+}
+
+test("navigate to project", async ({ page }) => {
+  await page.goto(sourceBrowsingFixture);
+
+  // Header.
+  {
+    const name = page.locator("text=source-browsing");
+    const urn = page.locator(
+      "text=rad:git:hnrkgd7sjt79k4j59ddh11ooxg18rk7soej8o",
+    );
+    const description = page.locator(
+      "text=Git repository for source browsing tests",
+    );
+
+    await expect(name).toBeVisible();
+    await expect(urn).toBeVisible();
+    await expect(description).toBeVisible();
+  }
+
+  // Project menu shows default selected branch and commit and contributor counts.
+  {
+    await expect(page.getByTitle("Current branch")).toContainText(
+      "main 530aabd",
+    );
+    await expectCounts({ commits: 7, contributors: 1 }, page);
+  }
+
+  // Navigate to the project README.md by default.
+  await expect(page.locator(".file-name")).toContainText("README.md");
+
+  // Show a commit teaser.
+  await expect(page.locator("text=dd068e9 Add README.md")).toBeVisible();
+
+  // Show rendered README.md contents.
+  await expect(page.locator("text=Git test repository")).toBeVisible();
+});
+
+test("show source tree at specific revision", async ({ page }) => {
+  await page.goto(sourceBrowsingFixture);
+  await page.locator('role=button[name="Commit count"]').click();
+
+  await page
+    .locator(".commit-teaser", { hasText: "335dd6d" })
+    .getByTitle("Browse the repository at this point in the history")
+    .click();
+
+  await expect(page.getByTitle("Current branch")).toContainText(
+    "335dd6dc89b535a4a31e9422c803199bb6b0a09a",
+  );
+  expect(page.locator(".source-tree")).toHaveText("bin/ src/");
+  await expectCounts({ commits: 2, contributors: 1 }, page);
+});
+
+test("source file highlighting", async ({ page }) => {
+  await page.goto(sourceBrowsingFixture);
+
+  await page.getByText("src/").click();
+  await page.getByText("true.c").click();
+
+  await expect(page.getByText("return")).toHaveCSS(
+    "color",
+    "rgb(180, 142, 173)",
+  );
+});
+
+test("navigate deep file hierarchies", async ({ page }) => {
+  await page.goto(sourceBrowsingFixture);
+
+  const sourceTree = page.locator(".source-tree");
+
+  await sourceTree.getByText("deep/").click();
+  await sourceTree.getByText("directory/").click();
+  await sourceTree.getByText("hierarchy/").click();
+  await sourceTree.getByText("is/").click();
+  await sourceTree.getByText("entirely/").click();
+  await sourceTree.getByText("possible/").click();
+  await sourceTree.getByText("in/").nth(1).click();
+  await sourceTree.getByText("git/").click();
+  await sourceTree.getByText("repositories/").click();
+  await sourceTree.getByText(".gitkeep").click();
+  await expect(
+    page.locator("text=0801ace Add a deeply nested directory tree"),
+  ).toBeVisible();
+
+  // After a page reload the tree browser is still expanded and we're still
+  // showing the .gitkeep file.
+  {
```

Comment:
Out of curiosity, I would like to know why we need to change the scope of the `page.reload() and the verification of the reloaded tree`. Couldn't we keep it in the same scope as the initial test?

Reply:
I added scopes is only to make larger tests more readable by dividing them up in visual chunks.

Reply:
nice, understood and I agree